react-components/src/architecture/concrete/boxDomainObject/MeasureLineView.ts

```diff
@@ -82,7 +94,57 @@ export class MeasureLineView extends GroupThreeView {
     if (this.domainObject.focusType === FocusType.Pending) {
       return undefined; // Should never be picked
     }
-    return super.intersectIfCloser(intersectInput, closestDistance);
+    // Implement the intersection logic here, because of bug in tree.js
+    const { domainObject, style } = this;
+    const radius = getRadius(domainObject, style);
+    if (radius <= 0) {
+      return;
+    }
+    const { points } = domainObject;
+    const { length } = points;
+    if (length < 2) {
+      return undefined;
+    }
+    // Just allocate all needed objects once
+    const prevPoint = new Vector3();
+    const thisPoint = new Vector3();
+    const intersection = new Vector3();
+
+    const radiusSquared = square(1.5 * radius); // Add 50% more to make it easier to pick
+    const ray = intersectInput.raycaster.ray;
+    const closestFinder = new ClosestGeometryFinder<DomainObjectIntersection>(ray.origin);
+
+    // TODO: The line below will case a tiny bug. The best is that the main intersection algorithm
+    // in the vieweer intersects the objects with depthTest == false first, before any other object and
+    // returns out if any of those objects are intersected. Same for Boxes. Now the intersection is somewhat arbitrarly
+    // if style.depthTest == false. Then the last one added to the viewer will be picked first,
+    // regardless of the distance to the mouse.
+    if (style.depthTest && closestDistance !== undefined) {
+      closestFinder.minDistance = closestDistance;
+    }
+    const loopLength = domainObject.measureType === MeasureType.Polygon ? length + 1 : length;
+    for (let i = 0; i < loopLength; i++) {
+      thisPoint.copy(points[i % length]);
+      thisPoint.applyMatrix4(CDF_TO_VIEWER_TRANSFORMATION);
```

Comment:
Are the points stored in CDF-coordinates?

Reply:
Yes